web/src/timerender.ts

```diff
@@ -423,17 +423,28 @@ export function get_full_time(timestamp: number): string {
     return formatISO(timestamp * 1000);
 }
 
+function get_current_time_to_hour(): Date {
+    const timestamp = new Date();
+    timestamp.setMinutes(0, 0);
+    return timestamp;
+}
+
 export function get_timestamp_for_flatpickr(timestring: string): Date {
     let timestamp;
+
+    // If timestring is undefined, return current time to the hour.
+    if (!timestring) {
```

Comment:
Don’t write ambiguous falsiness checks. Write `=== ""` or `=== undefined`.

If it’s really possible for this to be `undefined`, then the type signature is wrong and needs to be fixed.

Reply:
Sure, fixed it. `timestring=undefined` however, is only happening in case of `compose_setup.js`. So, maybe another option could be updating it there itself to make sure the `timestring` is rightly set, in case it is certain that we won't be using the method without any parameter in future.